jupyterlab_bigquery/src/components/list_items_panel/list_tree_item.tsx

```diff
@@ -409,18 +409,24 @@ export class DatasetResource extends Resource<DatasetProps> {
     };
     try {
       this.setState({ loading: true });
-      await bigQueryService
+      const tablesResult =
+        bigQueryService
         .listTables(dataset.projectId, dataset.name)
         .then((data: Dataset) => {
           newDataset.tables = data.tables;
           newDataset.tableIds = data.tableIds;
         });
-      await bigQueryService
+
+      const modelsResult =
+        bigQueryService
         .listModels(dataset.projectId, dataset.name)
         .then((data: Dataset) => {
           newDataset.models = data.models;
           newDataset.modelIds = data.modelIds;
         });
+
+      await tablesResult;
```

Comment:
You could also do `await Promise.all([tablesResult, modelsResult])`. Regardless of whether you use the existing code or that, you may want to add `.catch(` handlers to each Promise to not throw an exception if the list request fails.